chains/ethereum/writer_methods.go

```diff
@@ -244,50 +253,110 @@ func (w *writer) watchThenExecute(m msg.Message, data []byte, dataHash [32]byte,
 // voteProposal submits a vote proposal
 // a vote proposal will try to be submitted up to the TxRetryLimit times
 func (w *writer) voteProposal(m msg.Message, dataHash [32]byte) {
+	itxFailures := 0
 	for i := 0; i < TxRetryLimit; i++ {
 		select {
 		case <-w.stop:
 			return
 		default:
-			err := w.conn.LockAndUpdateOpts()
-			if err != nil {
-				w.log.Error("Failed to update tx opts", "err", err)
-				continue
-			}
-			// These store the gas limit and price before a transaction is sent for logging in case of a failure
-			// This declaration is necessary as tx will be nil in the case of an error when sending VoteProposal()
-			// We must also declare variables instead of using w.conn.Opts() directly as the opts are currently locked
-			// here but for all the logging after line 272 the w.conn.Opts() is unlocked and could be changed by another process
-			gasLimit := w.conn.Opts().GasLimit
-			gasPrice := w.conn.Opts().GasPrice
-
-			tx, err := w.bridgeContract.VoteProposal(
-				w.conn.Opts(),
-				uint8(m.Source),
-				uint64(m.DepositNonce),
-				m.ResourceId,
-				dataHash,
-			)
-			w.conn.UnlockOpts()
-
-			if err == nil {
-				w.log.Info("Submitted proposal vote", "tx", tx.Hash(), "src", m.Source, "depositNonce", m.DepositNonce, "gasPrice", tx.GasPrice().String())
-				if w.metrics != nil {
-					w.metrics.VotesSubmitted.Inc()
+			if w.conn.ItxClient() != nil && w.forwarderClient != nil && itxFailures < ItxRetryLimit {
```

Comment:
Having the check for `itxFailures < ItxRetryLimit` with the tx retry loop might cause misconceptions for someone running the bridge and using ITX of what the actual `TxRetryLimit` will be. The real `TxRetryLimit` in this case is `TxRetryLimit - ItxRetryLimit`. There is discussion of also having the retry limit and retry interval be config options to allow for more control. For someone using ITX the `TxRetryLimit` isn't the largest concern, as it should hopefully not need to fallback to the normal method of sending a proposal, however a config change could lead to confusions.

I was just thinking it may be cleaner to move the check for `w.conn.ItxClient() != nil && w.forwarderClient != nil` outside of this loop, and rather call respective methods with their own loops (`for i := 0; i < TxRetryLimit; i++ {...}` and `for i := 0; i < ItxRetryLimit; i++ {...}`). What do you think?

Reply:
Perhaps an additional ITX gas tank check would also be useful too, as to not have unnecessary retries

Reply:
Well spotted. I'll change the for loops. I also agree that a balance check would be nice - I'll add that

Reply:
In the end, instead of adding a balance check at the start I check for "Insufficient funds" returned from ITX and don't bother retrying if that occurs

Reply:
That should work, sounds good